### PR TITLE
Document Trainer Tower scripts

### PR DESCRIPTION
**Pull request by GriffinRichards**

### CODE REVIEW

File: include/renewable_hidden_items.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef GUARD_RENEWABLE_HIDDEN_ITEMS_H
 #define GUARD_RENEWABLE_HIDDEN_ITEMS_H
 
-bool32 sub_815D834(void);
+bool32 ReadTrainerHillAndValidate(void);
```
Comment (PikalaxALT) [inline include/renewable_hidden_items.h]: move this to cereader_tool?

Comment (GriffinRichards) [reply include/renewable_hidden_items.h]: Thats the header of the C file its defined in. Oddly it seems to be sandwiched between renewable items stuff, in emerald that function is in `ereader_helpers.c`

Comment (PikalaxALT) [reply include/renewable_hidden_items.h]: wdym? is the first function in that file i.e. the top of its .text section. move it to the end of cereader_tool.c

Comment (GriffinRichards) [reply include/renewable_hidden_items.h]: Oh whoops you're right, skimming the file I thought `sRenewableHiddenItems` was a function

File: src/fieldmap.c [hunk 2]
```diff
@@ -63,7 +63,7 @@ const struct MapHeader * mapconnection_get_mapheader(struct MapConnection * conn
     return Overworld_GetMapHeaderByGroupAndId(connection->mapGroup, connection->mapNum);
 }
 
-void not_trainer_hill_battle_pyramid(void)
+void not_trainer_tower_battle_pyramid(void)
```
Comment (PikalaxALT) [inline src/fieldmap.c]: Battle Pyramid didn't exist in FRLG.

Comment (GriffinRichards) [reply src/fieldmap.c]: Correct. The name just isn't synced, it doesnt have anything to do with Trainer Tower either.

**Pull request merged**